packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -491,6 +500,14 @@ class AndroidCameraCameraX extends CameraPlatform {
   /// Sets the flash mode for the selected camera.
   @override
   Future<void> setFlashMode(int cameraId, FlashMode mode) async {
+    CameraControl? cameraControl;
+    // Turn off torch mode if it is enabled and not being redundantly set.
+    if (mode != FlashMode.torch && torchEnabled) {
+      cameraControl = await camera!.getCameraControl();
+      await cameraControl.enableTorch(false);
```

Comment:
it still might be worth adding a note to the documentation here that points out the difference between this method and the underlying CameraX apis (specifically just that torch and flash can't be set independently in this plugin, like they can with the CameraX apis).

So something like
```
setFlashMode(id, FlashMode.torch)
takePicture(id)
setFlashMode(id, FlashMode.auto)
takePicture(id)
```
would behave differently in the plugin (torch disabled, flash mode on auto) from what someone familiar with the native apis would expect (**torch enabled**, flash mode on auto).